src/page-multi-facility/ReferenceDataModal/SyncNoUserFacilities.tsx

```diff
@@ -0,0 +1,319 @@
+  letter-spacing: -0.02em;
+  line-height: 150%;
+`;
+
+const CardDescription = styled.div`
+  margin: 20px 0px;
+`;
+
+const CardNavigation = styled.div`
+  border-top: 0.5px solid ${Colors.darkGray};
+  display: flex;
+  justify-content: space-between;
+  flex-direction: row-reverse;
+  padding-top: 20px;
+  width: 100%;
+`;
+
+const StateAndSystemForm = styled.div`
+  display: flex;
+  margin-bottom: 20px;
+`;
+
+type StateNameProps = {
+  stateName?: string;
+  setStateName: (stateName?: string) => void;
+};
+
+type SystemTypeProps = {
+  systemType?: string;
+  setSystemType: (systemType?: string) => void;
+};
+
+type CardStateProps = {
+  setActiveStep: (activeStep: string) => void;
+  setModalTitle: (modalTitle: string) => void;
+};
+
+type StateNameSelectionProps = StateNameProps & {
+  stateNames: string[];
+};
+
+const StateNameSelection: React.FC<StateNameSelectionProps> = (props) => {
+  const { stateName, stateNames, setStateName } = props;
+
+  return (
+    <InputSelect
+      label="State"
+      value={stateName}
+      onChange={(event) => {
+        setStateName(event.target.value);
+      }}
+    >
+      <option value="" />
+      {stateNames.map((name) => (
+        <option key={name} value={name}>
+          {name}
+        </option>
+      ))}
+    </InputSelect>
+  );
+};
+
+type StateAndSystemProps = StateNameProps & SystemTypeProps & CardStateProps;
+
+const StateAndSystemSelectionCard: React.FC<StateAndSystemProps> = (props) => {
+  const {
+    stateName,
+    setStateName,
+    systemType,
+    setSystemType,
+    setActiveStep,
+    setModalTitle,
+  } = props;
+
+  // This card doesn't have a title so we just clear out any existing
+  // title that may exist if the user navigates back to this card.
+  setModalTitle("");
+
+  const { data: localeDataSource } = useLocaleDataState();
+
+  const stateNames = Array.from(localeDataSource.keys()).filter(
+    (name) => name !== "US Total",
+  );
+
+  const handleNextButtonClick = async () => {
+    setActiveStep(SYNC_REFERENCE_FACILITIES_CARD);
+  };
+
+  return (
+    <CardContent>
+      <CardDescription>
+        Recidiviz provides a forecast of likely COVID-19 cases in your
+        facilities based on policies and populations.
+      </CardDescription>
+      <StateAndSystemForm>
+        <StateNameSelection
+          stateName={stateName}
+          stateNames={stateNames}
+          setStateName={setStateName}
+        />
+        <SystemTypeSelection
+          systemType={systemType}
+          setSystemType={setSystemType}
+        />
+      </StateAndSystemForm>
+      <CardNavigation>
+        <InputButton
+          styles={{
+            width: "80px",
+          }}
+          label="Next"
+          onClick={handleNextButtonClick}
+        />
+      </CardNavigation>
+    </CardContent>
+  );
+};
+
+type SyncReferenceFacilitiesCardProps = CardStateProps & {
+  stateName: StateNameProps["stateName"];
+  systemType: SystemTypeProps["systemType"];
```

Comment:
Is there a way to let TypeScript know that these values are required so that I don't have to add the `null` checks below?

Reply:
Looks like I can update this to:

```javascript
type SyncReferenceFacilitiesCardProps = CardStateProps & Required<{
  stateName: StateNameProps["stateName"];
  systemType: SystemTypeProps["systemType"];
}>;
```

Note the `Required` "wrapper"

Reply:
Hmm... this doesn't actually seem to do anything 🤷

Reply:
`Required` will just require that the prop is explicitly present when you call the component, but it won't prevent the values from being something falsy; i.e. if `StateNameProps["stateName"]` can be `null` than this type will still accept an explicitly passed `null`.

Maybe what you want is something more like `Pick<Required<StateNameProps & SystemTypeProps>, "stateName" | "systemType">` ... kind of a mouthful but I _think_ it will pass through the value types but make them un-nullable

Reply:
really I should have said `undefined` here, not null ... but hopefully my point still stands